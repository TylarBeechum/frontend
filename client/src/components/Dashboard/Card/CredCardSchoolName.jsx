import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const CredCardSchoolName = ({ credName, criteria }) => {
  return (
    <SchoolNameContainer>
      <h2>{credName}</h2>
      <p>{criteria}</p>
    </SchoolNameContainer>
  );
};

CredCardSchoolName.propTypes = {
  credName: PropTypes.string.isRequired,
  criteria: PropTypes.string.isRequired
};

const SchoolNameContainer = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: column;
  width: 45%;
`;

export default CredCardSchoolName;
