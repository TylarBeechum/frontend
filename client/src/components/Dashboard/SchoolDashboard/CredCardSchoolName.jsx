import React from 'react';
import styled from 'styled-components';

const CredCardSchoolName = ({ credName }) => {
  return (
    <SchoolNameContainer>
      <h2>{credName}</h2>
      <p>Credential description here. Lorem Ipsomething something idk</p>
    </SchoolNameContainer>
  );
};

const SchoolNameContainer = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: column;
`;

export default CredCardSchoolName;
