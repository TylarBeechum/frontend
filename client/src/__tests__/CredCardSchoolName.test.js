/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import {
  renderWithRouterAndProviders,
  rendererWithRouterAndProviders
} from '../__testHelpers__';

import CredCardSchoolName from '../components/Dashboard/Card/CredCardSchoolName';

describe('<CredCardSchoolName />', () => {
  describe('', () => {
    it('tests that the component does not crash when it is rendered', () => {
      const helpers = renderWithRouterAndProviders(
        <CredCardSchoolName credName="testName" criteria="testing" />
      );
    });
  });

  describe('should match the previous snapshot of the component', () => {
    it('tests that the component has not changed and matched the previous snapshot', () => {
      const tree = rendererWithRouterAndProviders(
        <CredCardSchoolName credName="testName" criteria="testing" />
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe('should render a the required strings successfully to the screen', () => {
    it('tests that the credential name string is rendered successfully', () => {
      const { getByText } = renderWithRouterAndProviders(
        <CredCardSchoolName credName="testName" criteria="testing" />
      );
      const credName = getByText(/testName/i);
      const criteria = getByText(/testing/i);
      expect(credName).toBeDefined();
    });

    it('tests that the criteria string is rendered successfully', () => {
      const { getByText } = renderWithRouterAndProviders(
        <CredCardSchoolName credName="testName" criteria="testing" />
      );
      const criteria = getByText(/testing/i);
      expect(criteria).toBeDefined();
    });
  });
});
