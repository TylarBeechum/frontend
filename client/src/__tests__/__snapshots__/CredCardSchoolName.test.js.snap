// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CredCardSchoolName /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 iDwiuh"
>
  <div
    className="sc-EHOje gwsCUe"
  >
    <h2>
      testName
    </h2>
    <p>
      testing
    </p>
  </div>
</div>
`;
